### PR TITLE
remove redundant methods in RichProgressBar

### PR DESCRIPTION
**Pull request by quancs**
## What does this PR do?

after #10896 `_init_progress` should run in the spawned processes already. We can remove this setstate method entirely.
see https://github.com/PyTorchLightning/pytorch-lightning/pull/10875#discussion_r768676725

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by awaelchli** [comment]
This PR does not address the comment you are linking to. There must be a misunderstanding. Two scenarios:
1. Either the Console object is pickleable and you can construct it on the main process, pass it in to the callback,
2. Or, the Console object is not pickleble, hence we only store the kwargs and make sure we construct it only in the worker process

Only one of the two applies, and given the existing code I suspect 2 is the case.

**Comment by quancs** [comment]
> This PR does not address the comment you are linking to. There must be a misunderstanding. Two scenarios:
> 
>     1. Either the Console object is pickleable and you can construct it on the main process, pass it in to the callback,
> 
>     2. Or, the Console object is not pickleble, hence we only store the kwargs and make sure we construct it only in the worker process
> 
> 
> Only one of the two applies, and given the existing code I suspect 2 is the case.

OK, then what should I do in this PR

**Comment by awaelchli** [comment]
I believe just removing this code:
```py

    def __getstate__(self):
        # can't pickle the rich progress objects
        state = self.__dict__.copy()
        state["progress"] = None
        state["_console"] = None
        return state
    def __setstate__(self, state):
        self.__dict__ = state
        self._console = Console(**self._console_kwargs)
```

File: pytorch_lightning/callbacks/progress/rich_progress.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import copy
```
Comment (quancs) [inline pytorch_lightning/callbacks/progress/rich_progress.py:14]:
```suggestion
```

**Comment by quancs** [comment]
@awaelchli removed.

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**